### PR TITLE
enhancement(kafka source): include kafka metadata as optional keys

### PR DESCRIPTION
**Pull request by spencergilbert**
Fixes: #1150
Fixes: #5143 
Fixes: #800

Modelled after the existing `key_field` configuration I inserted the topic, partition, and offset provided by rdkafka into events when configured. These should be opt-in and not affect users unless explicitly configured.

### CODE REVIEW

File: docs/reference/components/sources/kafka.cue [hunk 1]
```diff
@@ -83,6 +83,36 @@ components: sources: kafka: {
...
 			}
 		}
+		topic_key: {
+			common:      true
+			description: "The log field name to use for the Kafka topic key. If unspecified, the key would not be added to the log event."
+			required:    false
+			warnings: []
+			type: string: {
+				default: null
+				examples: ["topic_key"]
```
Comment (jszwedko) [inline docs/reference/components/sources/kafka.cue:93]: I might just use `topic`, `partition`, etc. as the examples. I think `key_field`'s example was `message_key` because `key` could be ambiguous (I guess?).

**Review by jszwedko — COMMENTED** [review]
Nice work! This looks good to me.

Could you add a few more assertions to the `kafka_source_consume_event` test to validate that these fields are set correctly?

File: docs/reference/components/sources/kafka.cue (hunk 1 above)
Comment (spencergilbert) [reply docs/reference/components/sources/kafka.cue:93]: Sounds good

**Comment by spencergilbert** [comment]
@jszwedko @jamtur01 @lukesteensen - I cleaned up the docs some and got the kafka integration test enabled

File: src/sources/kafka.rs [hunk 2]
```diff
@@ -272,14 +296,14 @@ mod integration_test {
...
-    use chrono::Utc;
+    use chrono::{SubsecRound, Utc};
     use rdkafka::{
         config::ClientConfig,
         producer::{FutureProducer, FutureRecord},
         util::Timeout,
     };
 
-    const BOOTSTRAP_SERVER: &str = "localhost:9092";
+    const BOOTSTRAP_SERVER: &str = "localhost:9091";
```
Comment (lukesteensen) [inline src/sources/kafka.rs:306]: Just curious, why this change?

Comment (spencergilbert) [reply src/sources/kafka.rs:306]: The makefile is starting kafka with a platintext listener on 9091, 9092 is SSL.

```
 587   │      -e KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://localhost:9091,SSL://localhost:9092,SASL_PLAINTEXT://localhost:9093 \
```

I suppose we could also look to setup vector with a cert and use 9092 :thinking:

Comment (jszwedko) [reply src/sources/kafka.rs:306]: I personally think PLAINTEXT is fine for this test but we could add other tests later for SSL.

**Review by jszwedko — APPROVED** [review]
This looks good to me! Nice work.

File: src/sources/kafka.rs (hunk 2 above)
Comment (spencergilbert) [reply src/sources/kafka.rs:306]: Probably worth another issue/pr to test the TLS config for the source

Comment (lukesteensen) [reply src/sources/kafka.rs:306]: Ah, that makes sense. Yeah, I'm fine to use plaintext for most tests and just have one that exercises TLS.

**Review by lukesteensen — APPROVED** [review]

**Comment by binarylogic** [comment]
Thanks @spencergilbert.

**Comment by jamtur01** [comment]
giphub
